Makefile.core.mk

```diff
@@ -72,8 +72,21 @@ JUNIT_REPORT := $(shell which go-junit-report 2> /dev/null || echo "${ISTIO_BIN}
 ISTIO_SERVE_DOMAIN ?= localhost
 export ISTIO_SERVE_DOMAIN
 
-ifeq ($(CONTEXT),production)
-baseurl := "$(URL)"
+# Determine current Git branch
+CURRENT_BRANCH := $(shell git rev-parse --abbrev-ref HEAD)
+
+# If the current branch is master, we use preliminary.istio.io/latest as the base URL
+ifeq ($(CURRENT_BRANCH),master)
+  baseurl := https://preliminary.istio.io/latest
+
+# If the current branch is release branch, we use the istio.io/latest as the base URL
+else ifneq ($(filter release-%, $(CURRENT_BRANCH)),)
+  baseurl := https://istio.io/latest
+
+# If the current branch is not master or release branch, we use /latest as the base URL
+# This is useful for feature branches or pull requests and deploy-preview
+else
+  baseurl := /latest
```

Comment:
I would still like to get fully qualified URLs in previews if possible. 
Is there a variable we can add here?